content/blog/nats-server-29-release.md

```diff
@@ -0,0 +1,489 @@
+
+```go
+sub, _ := js.SubscribeSync("events.*")
+// Do some work...
+sub.Unsubscribe()
+```
+
+The client library handles:
+
+- Looking up the stream by pattern matching the subject being subscribed to
+- Auto-creating an *unnamed* consumer (which makes an ephemeral)
+- Initializing a subscription and binding it to the consumer
+
+From the server's persective, it keeps track of whether at least one subscription is bound to this ephemeral consumer _and_ if the subscription is still _active_. If/when it detects the subscription is no longer active, it will automatically delete the ephemeral consumer after the [`InactiveThreshold`](https://docs.nats.io/nats-concepts/jetstream/consumers#inactivethreshold) period (which defaults to five seconds).
+
+In contrast, durables are intended to survive process crashes, restarts (client or server), and regardless if there are any active subscriptions. The lifecycle of a durable is expected to be managed by the application/operator.
+
+However, what happens if consumers are being created and not properly cleaned up? Each consumer has a small amount of state associated with it that the server needs to keep track of, replicate, restore, etc. To prevent an evergrowing number of consumers, this consumer configuration can now be applied to durables.
+
+```go
+js.AddConsumer("EVENTS", &nats.ConsumerConfig{
+  Durable: "event-processor",
+  AckPolicy: nats.AckExplicitPolicy,
+  InactiveThreshold: time.Hour,
+})
+```
+
+In practice, an inactive threshold for a durable should be set to a value that would be highly unlikely the consumer would *not* have been active in the meantime. In other words, if the consumer is expected to be active regularly every couple minutes, if there is no activity for an hour, that may be a good indicator that the consumer can be auto-deleted.
+
+## Scale and Mobility
+
+NATS was designed to scale in multiple dimensions whether the need is high message throughput or geographically distributed nodes to service users or devices.
+
+Combine this with [subject-based messaging][subject-messaging] and intelligent routing of client messages to the closest _servicer_,  NATS provides a global, location transparent connective _fabric_ for messaging, streams, key-value stores, and object storage buckets.
+
+This release brings _more_ capabilities to scale in more dimensions and make data locality more transparent.
+
+[subject-messaging]: https://docs.nats.io/nats-concepts/subjects
+
+#### Replica and mirror-based direct gets
+
+Streams have supported a `GetMsg` operation since JetStream was first released. This method is avaliable on the client's JetStream Manager interface and takes a stream name and the specific sequence number to get a message from the stream. For example, in Go:
```

Comment:
```suggestion
Streams have supported a `GetMsg` operation since JetStream was first released. This method is available on the client's JetStream Manager interface and takes a stream name and the specific sequence number to get a message from the stream. For example, in Go:
```